projectroles/forms.py

```diff
@@ -541,9 +554,80 @@ def __init__(self, project=None, current_user=None, *args, **kwargs):
             self.fields['description'].widget = forms.HiddenInput()
             self.fields['readme'].widget = forms.HiddenInput()
 
+    @classmethod
+    def _app_settings_validation(
+        self,
+        cleaned_data,
+        app_plugins,
+        app_settings,
+        p_kwargs,
+        instance,
+        instance_owner_as,
+    ):
+        """Validate and clean app_settings form fields"""
+        errors = []
+        if cleaned_data.get('type') == PROJECT_TYPE_PROJECT:
```

Comment:
To avoid extreme indentation, we could also just return here with `cleaned_data` if type is `PROJECT_TYPE_CATEGORY`.